src/Controller/AppController.php

```diff
@@ -323,12 +317,14 @@ public function getSupportedLanguage()
             $supportedLanguages[$browserCompatibleCode] = $langs[0];
         }
 
-        if (isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
+        $header = $this->request->getHeaderLine('Accept-Language');
+        if (!empty($header)) {
 
-            $browserLanguages = explode(',', $_SERVER['HTTP_ACCEPT_LANGUAGE']);
+            $browserLanguages = explode(',', $header);
 
             foreach ($browserLanguages as $browserLang) {
                 $browserLangArray = explode(';', $browserLang);
```

Comment:
There's also [`Cake\Http\ServerRequest::acceptLanguage()`](https://api.cakephp.org/3.7/class-Cake.Http.ServerRequest.html#_acceptLanguage).

For a header line like `Accept-Language: en,de-AT;q=0.8,en-US;q=0.5,fr-FR;q=0.3` it would return
```
2020-09-18 13:16:43 Debug: Array
(
    [0] => en
    [1] => de-at
    [2] => en-us
    [3] => fr-fr
)
```

Reply:
Right! :smile: